### ISSUE TITLE
Will RepCut support higher version of FIRRTL?

### ISSUE THREAD

**Issue by poppindouble** [ucsc-vama/essent#24]
I noticed current RepCut only supports FIRRTL with 1.4.3
```
libraryDependencies += "edu.berkeley.cs" %% "firrtl" % "1.4.3"
```
while Essent supports FIRRTL with FIRRTL 1.5.6
```
libraryDependencies += "edu.berkeley.cs" %% "firrtl" % "1.5.6"
```

I was trying to test RepCut with RISCV-BOOM designs other than current artifacts from Zenodo.
The latest version of Chipyard which supports FIRRTL 1.4.3 is 1.4.0. However, Chipyard 1.4.0 has a lot of trouble producing the FIR files ([ref1](https://github.com/ucb-bar/chipyard/issues/818
), [ref2](https://github.com/ucb-bar/chipyard/issues/1003)).

**Comment by haoozi** [comment]
Yes we do have plan to support higher FIRRTL version. In fact, the patch that bumps up FIRRTL on Essent (patch #21 ) should also works for repcut branch. This patch is not integrated into repcut branch yet but can be merged by:
```
git checkout repcut
# merge commit 18dd3d5a7444706ae1397f2bf66a61312e64beb8
git cherry-pick 18dd3d5a7444706ae1397f2bf66a61312e64beb8 --no-commit
# update sbt version
echo "sbt.version=1.9.6" > ./project/build.properties
```
This patch bumps FIRRTL to 1.5.6. You should be able to build by `sbt assembly`.

FIRRTL 1.5.6 is the last stable version of Scala FIRRTL compiler. Since Chisel 3.6.0 the backend moved to CIRCT,  deprecating Scala compiler (See release note of FIRRTL: https://github.com/chipsalliance/firrtl). As the result, we won't be able to bump to higher FIRRTL. 1.5.6 is likely to be the version we will stay for long term. 

If you wish to run Chipyard design please also modify the TestHarness or ChipTop module, remove analog pins and avoid multiple clock domain. They are not supported.

**Comment by sbeamer** [comment]
We have now updated it to support FIRRTL 1.6.1. Due to change of backend (within Chisel/FIRRTL), we do not anticipate updating beyond that. This version does support Scala 2.13.18, which is supported JDK 26 (and before), so this code will executable for some time.
